### DISCUSSION TITLE
Help with Firefox profiles in Remote Desktop environment

### DISCUSSION THREAD

**Discussion by eric-templin** [mozilla/policy-templates#1116] [Q&A]
I've come into an environment where Firefox is installed in an RDS farm environment that uses roaming profiles, and the hosts keep running low on free space.

What I'm seeing is roaming profiles that contain Firefox profiles (AppData roaming) in excess of 5GB per user.  It was my understanding that this would only contain things like saved passwords, bookmarks, etc.?  AppData Local (where the "cache" would be) doesn't seem to be an issue as that gets wiped at logoff.  I'm trying to find some ways to either systematically limit the size or number of these roaming profiles, or to programmatically clean them up without breaking the user.

My main question is... if the user's profile.ini file lists 5 different profile folders, but the Firefox\Profiles directory contains dozens (and in some cases HUNDREDS - yes really) of profile folders, is it safe to delete the folders that are NOT listed in the profile.ini file?  All of these "old" folders contain about 50MB of files each, and they add up quickly since there's no self-cleanup.

On a similar topic, what scenarios would cause a user's profile.ini file to contain several dozen profile entries? I can't imagine these users are creating their own new profiles.

The RDS servers do receive regular Firefox patches, though they may be off a version between them depending on the patching cadence. Not sure its related but thought I would mention it.

**Comment by mkaply** [comment]
Yes, it is safe to delete folders that are not in profiles.ini

I'm wondering if somehow there are multiple Firefox versions in different places.

How many values that look like:

[InstallSOMEHEXNUMBER]

do you have in the profile?

My guess is what is eating up space is the storage directory.

See:

https://bugzilla.mozilla.org/show_bug.cgi?id=1395705

You might just delete the storage directory every often. Users will lose cookies and other site data, but this is similar to what Chrome/Edge do (they don't roam site data)

**Reply by eric-templin** [reply]
Attached is the profiles.ini file for an offending user.  It has 99 profile folders on disk, amounting to over 6GB of data that has to be synced at every logon/logoff.  Every single profile folder is referenced in the INI file, so there's none that I can just delete using that as a reference.

The RDS farm runs 24/7, and does receive monthly Firefox updates via SCCM/PatchMyPC.  Because it's a 24/7 operation, they can only patch one host server in each farm, and have to move users between the hosts to get all the servers patched each month.   So, a user could move from last month's version to this month's version and potentially back to the previous month's version before all the servers are at the current version.  I assume this would generate a new profile for each version leap forward.  Is one generated on a downgrade?

For the user whose INI I attached, there have been 7 profiles modified this month, 3 last month and 5 the month before, so maybe Firefox is also self updating in parallel with the deployed patches (I'll have to look into that).  The profiles go all the way back to the beginning of last year.

Assuming the servers get brought to the latest version once a month, even with the possibility of self-update in between, is it safe to merely purge any profile folders where no data in them has been modified in say the past 60 days?  This would significantly reduce the profile folder sizes across the board for us. 

There appears to be a script in that link you included about this issue.  I'll give that a try and see if that helps things out.   Doesn't look like it will clean up old profile folders but maybe I'll also test purging out anything older than say 60 days.

Thanks!

**Reply by eric-templin** [reply]
[profiles.ini.txt](https://github.com/user-attachments/files/15839171/profiles.ini.txt)

**Comment by mkaply** [comment]
You also might try setting

MOZ_ALLOW_DOWNGRADE 

in your environment and see if that helps.

This will keep Firefox from creating a new profile when a user moves between minor versions.

I assume Firefox is always installed in the same directory?

A couple interesting things from the profile. There are two installs of Firefox:

[InstallE7CF176E110C211B] and  [Install308046B0AF4A39CB]

[Install308046B0AF4A39CB] is the most recent one as it has profile-97

[InstallE7CF176E110C211B] is older and hasn't been used in quite a while (profile 55)

My guess is what is happening is new profiles are just getting created constantly because of version mismatches.

I definitley think you can delete older profiles

-97 seems to be the current one

**Reply by eric-templin** [reply]
Do you know if MOZ_ALLOW_DOWNGRADE can be set with Group Policy?  From what I'm reading it is passed in the command to launch Firefox?

**Comment by mkaply** [comment]
You can set it as a windows environment variable.

The only thing you can set with policy is LegacyProfiles which prevents Firefox installed in different directories from creating a new profile.
